internal/action/actions.go

```diff
@@ -1676,50 +1676,58 @@ func (h *BufPane) PageDown() bool {
 
 // SelectPageUp selects up one page
 func (h *BufPane) SelectPageUp() bool {
+	pageOverlap := int(h.Buf.Settings["pageoverlap"].(float64))
+	scrollAmount := h.BufView().Height - pageOverlap
 	if !h.Cursor.HasSelection() {
 		h.Cursor.OrigSelection[0] = h.Cursor.Loc
 	}
-	h.MoveCursorUp(h.BufView().Height)
+	h.MoveCursorUp(scrollAmount)
 	h.Cursor.SelectTo(h.Cursor.Loc)
+	h.ScrollUp(scrollAmount)
```

Comment:
Just like in `CursorPageUp/Down`, it should be done for the first cursor only, right? Otherwise `SelectPageUp/Down` selects multiple pages instead of one.

Reply:
Fixed.

Reply:
I was talking about `ScrollUp()`, not about `SelectTo()`.

Sorry for confusion, I said "selects multiple pages instead of one" but that's not the problem. It selects exactly one page for each cursor, as it should. The problem is with the view: it scrolls the view multiple times, so the resulting behavior is unpredictable (as you should notice if you test `SelectPageUp/Down` with multicursors): sometimes the view does not include the selection at all, sometime it includes the selection but still not nano-like at all.

Reply:
OK, fixed properly this time (I think).